relay-server/src/actors/project.rs

```diff
@@ -826,9 +909,18 @@ impl Project {
 
         match self.expiry_state() {
             // If the new state is invalid but the old one still usable, keep the old one.
-            ExpiryState::Updated(old) | ExpiryState::Stale(old) if state.invalid() => state = old,
+            ExpiryState::Updated(old) | ExpiryState::Stale(old)
+                if state.invalid() || state.project_id.is_none() =>
```

Comment:
need some critical eyes on this `state.project_id.is_none()` line.

currently we check after the aggregator if the project_id is there before ratelimiting, and if not, we send it back to the aggregator so that it may be tried again later.  
 
Now that the rate-limiting is entirely before the aggregator, we can no longer do that, so to avoid losing these buckets (or not rate-limiting them), I thought we could rather keep them in the pre-aggregator longer until we have a projectstate with a project_id. 

if anyone can think of an unintended bad consequence of this then let me know. I've looked through it but will check it more later.

Reply:
Hmm, not sure if this check is even necessary, will have to think about it a bit more.

Reply:
Sentry always sends a `project_id`, so I would rather not alter the logic of this check, and log an error on unexpected `None` in `rate_limit_and_merge_buckets`, as you are already doing.

https://github.com/getsentry/sentry/blob/ea1023245fa1962fc3989cb5bb016f6c70b3ea57/src/sentry/relay/config/__init__.py#L354

Reply:
right, so  I guess it's only lacking in static/proxy mode? it's a bit annoying that the type system doesnt take care of this, but yes then it makes sense to rather log an error

Reply:
Even in static mode we guarantee a project_id: https://github.com/getsentry/relay/blob/2a86991dfbdd2ae6b1dc2a188906947ba89f0f86/relay-server/src/actors/project_local.rs#L101-L108

Reply:
so given that it's an optional, when is it ever None?